go/pkg/client/tree.go

```diff
@@ -89,6 +155,20 @@ func loadFiles(execRoot string, excl []*command.InputExclusion, path string, fs
 			},
 		}
 		return nil
+	} else if t == command.SymlinkInputType {
+		fs[path] = &fileSysNode{
+			symlink: &symlinkNode{target: meta.Symlink.Target},
+		}
+		if meta.Symlink.IsDangling {
+			return nil
+		}
+		// If we can proceed here, we've already verified that target is
+		// under execRoot, so this should never fail.
+		relTarget, err := getTargetRelPath(execRoot, path, meta.Symlink.Target)
+		if err != nil {
+			panic(fmt.Sprintf("Invalid target=%q err=%v", relTarget, err))
+		}
+		return loadFiles(execRoot, excl, relTarget, fs, chunkSize, cache, opts)
```

Comment:
What happen if symlink target is already in other upload paths?

Reply:
It would result in a repeated read, which probably isn't a big deal? I.e. it will be no different from the case where we specify a normal file path, and a directory that contains that file.

Reply:
I see, but isolate/isolated don't upload target of symlink if it is not included in upload paths.
So I think it is fine to make whether upload target of symlink or not user's responsibility.

Reply:
> So I think it is fine to make whether upload target of symlink or not user's responsibility.

See https://github.com/bazelbuild/remote-apis-sdks/pull/229#discussion_r521422354? I added a `FollowsTarget` option. The SDK sets this to `true` by default. FYI @ola-rozenfeld

Reply:
Yep, makes sense, thank you!